File: Airship/AirshipCore/Source/common/UAPushReceivedEvent.m
/* Copyright Airship and Contributors */

#import "UAPushReceivedEvent+Internal.h"
#import "UAEvent+Internal.h"
#import "UAAnalytics+Internal.h"

@implementation UAPushReceivedEvent

+ (instancetype)eventWithNotification:(NSDictionary *)notification {
    NSMutableDictionary *data = [NSMutableDictionary dictionary];
    [data setValue:[notification objectForKey:kUAPushMetadata] forKey:@"metadata"];
    [data setValue:[notification objectForKey:@"_"] ?: kUAMissingSendID forKey:@"push_id"];

    UAPushReceivedEvent *event = [[self alloc] init];
    event.data = data;
    return event;
}

- (NSString *)eventType {
    return @"push_received";
}

@end

File: Airship/AirshipCore/Source/UAPushReceivedEvent.swift
/* Copyright Airship and Contributors */

/**
 * @note For Interrnal use only :nodoc:
 */
@objc
public class UAPushReceivedEvent : UAEvent {

    @objc
    public override var eventType : String {
        get {
            return "push_received"
        }
    }

    private let _data : [AnyHashable : Any]

    @objc
    public override var data: [AnyHashable : Any] {
        get {
            return self._data
        }
    }

    @objc
    public init(notification: [AnyHashable : Any]) {
        var data: [AnyHashable : Any] = [:]
        data["metadata"] = notification["com.urbanairship.metadata"]
        data["push_id"] = notification["_"] ?? "MISSING_SEND_ID"

        self._data = data

        super.init()
    }
}
